Classes/Service/Shopware/OrderClient.php

```diff
@@ -0,0 +1,133 @@
+ *  Copyright notice
+ *
+ *  (c) (c) 2017 Axel Boeswetter <boeswetter@portrino.de>, portrino GmbH
+ *
+ *  All rights reserved
+ *
+ *  This script is part of the TYPO3 project. The TYPO3 project is
+ *  free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  The GNU General Public License can be found at
+ *  http://www.gnu.org/copyleft/gpl.html.
+ *
+ *  This script is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+use TYPO3\CMS\Core\Utility\ArrayUtility;
+use TYPO3\CMS\Extbase\Persistence\ObjectStorage;
+
+/**
+ * Class OrderClient
+ *
+ * @package Portrino\PxShopware\Service\Shopware
+ */
+class OrderClient extends AbstractShopwareApiClient implements OrderClientInterface
+{
+
+    /**
+     * @param string $term
+     * @param int $limit
+     * @param bool $doCacheRequest
+     * @param array $params
+     * @return ObjectStorage
+     */
+    public function findByTerm($term, $limit = -1, $doCacheRequest = true, $params = [])
+    {
+        $shopwareModels = new ObjectStorage();
+
+        /**
+         * only search for id if term is integer
+         */
+        if (is_numeric($term)) {
+            $filter = [
+                [
+                    'property' => 'id',
+                    'expression' => '=',
+                    'value' => $term
+                ],
+                [
+                    'operator' => '1',
+                    'property' => 'number',
+                    'expression' => 'LIKE',
+                    'value' => '%' . $term . '%'
+                ],
+            ];
+        } else {
+            $filter = [
+                [
+                    'property' => 'customer.email',
+                    'expression' => 'LIKE',
+                    'value' => '%' . $term . '%'
+                ],
+                [
+                    'operator' => '1',
+                    'property' => 'customer.firstname',
+                    'expression' => 'LIKE',
+                    'value' => '%' . $term . '%'
+                ],
+                [
+                    'operator' => '1',
+                    'property' => 'customer.lastname',
+                    'expression' => 'LIKE',
+                    'value' => '%' . $term . '%'
+                ]
+            ];
+        }
+
+        ArrayUtility::mergeRecursiveWithOverrule($params, [
+            'limit' => $limit,
+            'sort' => [
+                [
+                    'property' => 'number',
```

Comment:
isn´t it better to sort by lastname?

Reply:
This is just for an autocomplete suggest wizard in the TYPO3 backend. The order ID field will be filled by API callbacks. This is just a convenience feature and the header of the autocomplete item is the order number. That's why it's sorted by order number.